src/metabase/models/recent_views.clj

```diff
@@ -155,80 +162,110 @@
 (defmulti fill-recent-view-info
   "Fills in additional information for a recent view, such as the display name of the object.
 
-  - When called from `GET /popular_items`, the `model_object` field will be present, and should be used instead of
-  querying the database for the object."
+  For most things, we try to gather the information in a single query, but for cards and datasets, but certain things
+  are way easier to check with code (e.g. a collection's parent collection.)"
   (fn [{:keys [model #_model_id #_timestamp card_type]}]
-    (or (get {"model" :dataset
-              "question" :card} card_type)
+    (or (get {"model" :dataset "question" :card} card_type)
         (keyword model))))
 
 (defmethod fill-recent-view-info :default [m] (throw (ex-info "Unknown model" {:model m})))
 
-(defn get-parent-coll
+(defn- get-parent-coll*
   "Gets parent collection info for a recent view item."
```

Comment:
I feel like this function name and docstring is a little misleading — I was expecting it to take a recent view item and return a collection, but it actually takes a collection ID and fetches its details?